### ISSUE TITLE
puppet source

### ISSUE THREAD

**Issue by otteydw** [voxpupuli/puppet-archive#151]
Does this module support a puppet:/// source?  If not, is it planned?
This source works in nanliu-staging.

I tried

```
    archive { "/usr/java/jdk-${jdk_version}-linux-x64.tar.gz":
        ensure          => present,
        extract         => true,
        extract_path    => '/usr/java',
        source          => "puppet:///files/java/jdk-${jdk_version}-linux-x64.tar.gz",
        creates         => "${jdk_dir}",
        cleanup         => true,
    }
```

Got this error:

Error: Failed to apply catalog: Parameter source failed on Archive[/usr/java/jdk-7u55-linux-x64.tar.gz]: invalid source url: puppet:///files/java/jdk-7u55-linux-x64.tar.gz at /etc/puppetlabs/puppet/manifests/boomi-jdk2.pp:67
Wrapped exception:
invalid source url: puppet:///files/java/jdk-7u55-linux-x64.tar.gz
Wrapped exception:
invalid source url: puppet:///files/java/jdk-7u55-linux-x64.tar.gz

**Comment by jyaworski** [comment]
@nanliu didn't this happen at some time?

@otteydw what version are you using? Of Archive, that is.

**Comment by otteydw** [comment]
@jyaworski - I am using puppet-archive 0.5.1 which I installed today.

**Comment by nanliu** [comment]
nope, it currently does not support puppet:///, you can use a file resource + archive with local path, but this won't allow cleanup. This is a new feature request.

**Comment by jhg03a** [comment]
@nanliu How would you go about this workaround?  There doesn't appear to be any documentation on local file archive usage.

**Comment by rnelson0** [comment]
The [duplicate test](https://github.com/voxpupuli/puppet-archive/blob/5e457c5af67681bfbcbd3d6bd3b58d06cd71f80d/tests/duplicate.pp) implies that `file://` will suffice.

**Comment by jhg03a** [comment]
This works for me.  However there was a lot of trial and error here.  There are many, many ways to get a silent failure that looks like success.

```
  $docs_filename = 'help.tar.gz'
  $docs_gz_path  = "/tmp/${docs_filename}"
  $homedir = '/home/myuser/'

  file {$docs_gz_path:
    ensure => file,
    source => "puppet:///modules/profile/${docs_filename}",
  }

  archive { $docs_gz_path:
    path          => $docs_gz_path,
    extract       => true,
    extract_path  => $homedir,
    creates       => "${homedir}/help" #directory inside tgz
    require       => [ File[$homedir], File[$docs_gz_path] ],
  }
```

**Comment by rnelson0** [comment]
Would you be willing to open a PR to update the documentation, to help others avoid this issue?

**Comment by jhg03a** [comment]
Done.

**Comment by nanliu** [comment]
If you have examples for those silent errors, please file an issue.

**Comment by jhg03a** [comment]
They are mostly surrounding proper usage.  For instance: use source instead of path or leave off the creates.  The catalog will compile and run successfully, but no useful output will occur.

**Comment by nanliu** [comment]
This feature has been added and released in 2.0.0
